podcast/2.17.md

```diff
@@ -0,0 +1,487 @@
+
+**Max Brunsfeld:**
+
+Right. Yeah. It and Operational Transformation are like the two kinds of approaches and CRDTs require less central coordination than Operational Transformation. And there's been all this work in recent times to make them efficient enough to use all the time in an editor. Yeah, after GitHub, Antonio went and worked out of this company called Ditto. That was all about data replication and he learned even more about CRDTs. We're trying to take that, take what we learned there, and also take what we learned about building a fast text editor. Something we all kind of became passionate about when working on Atom was wanting to make an editor that was responsive all the time and feels like it's never in your way to use. And to build an editor that sort of lets you collaborate on code in a way that we have not yet been able to do.
+
+So, there's some common features that exist now in other tools but I want to come in, in the morning, when I come to work, and not necessarily open up Slack or GitHub, but open up my editor and see who's there. I also want to see what conversation has taken place and then be able to look through that conversation and see where in the code it applies. I want to click on what somebody said and be there in the code at that moment that they said that at that time and also see where that code is now and how it's evolved since then.
+
+And be able to for myself, on the other flip side, like at the end of my day when I'm handing off to Antonio, rather than going over to Slack and typing this long thing and pasting back code snippets in Slack, I want to click on some code or go to some code that I'm talking about and just start talking. And that association between this code at that moment in time and what I'm saying to be captured in the tool, or be able to click on a compiler error and start talking about that.
+
+**Beyang Liu:**
+
+It's like you're creating this tour of some recently written code and being able to link dynamically into specific snippets. Is that kind of the idea?
+
+**Max Brunsfeld:**
+
+It's linked back and forth and in a way that's sort of useful in a real time situation where if we're there together, instead of typing in Slack about the code, we're typing in the code editor. We're also simultaneously making code changes, but also for conversations that happen at separate points in time. Like right now, if I want to find out about the history of a certain piece of code, I would go to GitHub and see these closed poll requests that now have discussions in them about that code at different points in time–but it's hard. Like I would go to the pull request and search for certain words that would come up in the discussion but I can't just go to that file and say, tell me the discussions that relate to this.
```

Comment:
```suggestion
It's linked back and forth and in a way that's sort of useful in a real time situation where if we're there together, instead of typing in Slack about the code, we're typing in the code editor. We're also simultaneously making code changes, but also for conversations that happen at separate points in time. Like right now, if I want to find out about the history of a certain piece of code, I would go to GitHub and see these closed pull requests that now have discussions in them about that code at different points in time–but it's hard. Like I would go to the pull request and search for certain words that would come up in the discussion but I can't just go to that file and say, tell me the discussions that relate to this.
```